cpp/tests/reductions/segmented_reduction_tests.cpp

```diff
@@ -387,6 +387,140 @@ TEST_F(SegmentedReductionTestUntyped, ReduceEmptyColumn)
+                                out_type,
+                                null_policy::EXCLUDE);
+
+    CUDF_TEST_EXPECT_COLUMNS_EQUAL(*res, expect);
+  }
+}
+
+TYPED_TEST(SegmentedReductionFixedPointTest, MinIncludeNulls)
+{
+  // scale: -2, 0, 5
+  // [1, 2, 3], [1, null, 3], [1], [null], [null, null], []
+  // values:    {1, 2, 3, 1, XXX, 3, 1, XXX, XXX, XXX}
+  // offsets:   {0, 3, 6, 7, 8, 10, 10}
+  // nullmask:  {1, 1, 1, 1, 0, 1, 1, 0, 0, 0}
+  // outputs:   {1, XXX, 1, XXX, XXX, XXX}
+  // output nullmask: {1, 0, 1, 0, 0, 0}
+
+  using RepType = device_storage_type_t<TypeParam>;
+
+  for (auto scale : {-2, 0, 5}) {
+    auto input     = fixed_point_column_wrapper<RepType>({1, 2, 3, 1, XXX, 3, 1, XXX, XXX, XXX},
+                                                     {1, 1, 1, 1, 0, 1, 1, 0, 0, 0},
+                                                     numeric::scale_type{scale});
+    auto offsets   = std::vector<size_type>{0, 3, 6, 7, 8, 10, 10};
+    auto d_offsets = thrust::device_vector<size_type>(offsets);
+    auto out_type  = column_view(input).type();
+    auto expect    = fixed_point_column_wrapper<RepType>(
+      {1, XXX, 1, XXX, XXX, XXX}, {1, 0, 1, 0, 0, 0}, numeric::scale_type{scale});
+
+    auto res = segmented_reduce(input,
+                                d_offsets,
+                                *make_min_aggregation<segmented_reduce_aggregation>(),
+                                out_type,
+                                null_policy::INCLUDE);
+
+    CUDF_TEST_EXPECT_COLUMNS_EQUAL(*res, expect);
+  }
+}
+
+TYPED_TEST(SegmentedReductionFixedPointTest, MinExcludeNulls)
+{
+  // scale: -2, 0, 5
+  // [1, 2, 3], [1, null, 3], [1], [null], [null, null], []
+  // values:    {1, 2, 3, 1, XXX, 3, 1, XXX, XXX, XXX}
+  // offsets:   {0, 3, 6, 7, 8, 10, 10}
+  // nullmask:  {1, 1, 1, 1, 0, 1, 1, 0, 0, 0}
+  // outputs:   {1, 1, 1, XXX, XXX, XXX}
+  // output nullmask: {1, 1, 1, 0, 0, 0}
+
+  using RepType = device_storage_type_t<TypeParam>;
+
+  for (auto scale : {-2, 0, 5}) {
+    auto input     = fixed_point_column_wrapper<RepType>({1, 2, 3, 1, XXX, 3, 1, XXX, XXX, XXX},
+                                                     {1, 1, 1, 1, 0, 1, 1, 0, 0, 0},
+                                                     numeric::scale_type{scale});
+    auto offsets   = std::vector<size_type>{0, 3, 6, 7, 8, 10, 10};
+    auto d_offsets = thrust::device_vector<size_type>(offsets);
+    auto out_type  = column_view(input).type();
+    auto expect    = fixed_point_column_wrapper<RepType>(
+      {1, 1, 1, XXX, XXX, XXX}, {1, 1, 1, 0, 0, 0}, numeric::scale_type{scale});
+
+    auto res = segmented_reduce(input,
+                                d_offsets,
+                                *make_min_aggregation<segmented_reduce_aggregation>(),
+                                out_type,
+                                null_policy::EXCLUDE);
+
+    CUDF_TEST_EXPECT_COLUMNS_EQUAL(*res, expect);
+  }
+}
+
```

Comment:
Should there be min/max tests with no nulls? Seems there is a kernel path not being tested?

Reply:
Good catch, test added.